lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import grp
+import json
+import os
+import pwd
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            bhard=dict(type='str'),
+            bsoft=dict(type='str'),
+            ihard=dict(type='int'),
+            isoft=dict(type='int'),
+            mountpoint=dict(type='str', required=True),
+            name=dict(type='str'),
+            rtbhard=dict(type='str'),
+            rtbsoft=dict(type='str'),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            type=dict(type='str', required=True, choices=['user', 'group', 'project'])
+        ),
+        supports_check_mode=True
+    )
+
+    quota_type = module.params['type']
+    name = module.params['name']
+    mountpoint = module.params['mountpoint']
+    bhard = module.params['bhard']
+    bsoft = module.params['bsoft']
+    ihard = module.params['ihard']
+    isoft = module.params['isoft']
+    rtbhard = module.params['rtbhard']
+    rtbsoft = module.params['rtbsoft']
+    state = module.params['state']
+
+    if bhard is not None:
+        bhard = human_to_bytes(bhard)
+
+    if bsoft is not None:
+        bsoft = human_to_bytes(bsoft)
+
+    if rtbhard is not None:
+        rtbhard = human_to_bytes(rtbhard)
+
+    if rtbsoft is not None:
+        rtbsoft = human_to_bytes(rtbsoft)
+
+    result = dict(
+        changed=False
+    )
+
+    if not os.path.ismount(mountpoint):
+        module.fail_json(msg='%s is not a mountpoint' % mountpoint, **result)
+
+    mp = get_fs_by_mountpoint(mountpoint)
+    if mp is None:
+        module.fail_json(msg='%s is not a mountpoint or not located on an xfs filesystem.' % mountpoint, **result)
+
+    if quota_type == 'user':
+        type_arg = '-u'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'uquota' not in mp['mntopts'] \
+                and 'usrquota' not in mp['mntopts'] \
+                and 'quota' not in mp['mntopts'] \
+                and 'uqnoenforce' not in mp['mntopts'] \
+                and 'qnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the uquota/usrquota/quota/uqnoenforce/qnoenforce option.'
+                    % mountpoint, **result
+            )
+        try:
+            pwd.getpwnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name, **result)
```

Comment:
Same here, we tend to quote.
```suggestion
            module.fail_json(msg="User '%s' does not exist." % name, **result)
```